src/index.ts

```diff
@@ -1,69 +1,69 @@
+type InjectableVueClass = VueConstructor<Vue> & {
+    options: ComponentOptions<Vue>
 };
 
-const VueInversify: PluginFunction<any> = instance => {
+const VueInversify: PluginFunction<unknown> = instance => {
     instance.mixin({
         beforeCreate() {
-            if (this.$options.container) 
-                this.$container = this.$options.container;
-            else {
-                // Cascade the '$container' down all children as they're created
-                const root = (this.$parent || this);
-                this.$container = root.$container;
-            }
-
-            const $inject = (this.constructor as InjectableVueClass).$inject;
-            const container = this.$container;
+            const dependencies = this.$options.dependencies;
+            const container = this.$options.container || (this.$options.container = getContainer(this));
             
-            if ($inject) {
+            if (dependencies) {
                 if (!container)
                     throw new TypeError('vue-inversify: encountered props marked with @inject but no container was provided when creating the Vue instance');
 
-                Object.keys($inject).forEach(key => {
+                Object.keys(dependencies).forEach(key => {
                     Object.defineProperty(this, key, {
                         enumerable: true,
-                        value: container.get($inject[key])
+                        value: container.get(dependencies[key])
                     });
                 });
             }
         }
     });
 };
 
-const inject = (identifier?: interfaces.ServiceIdentifier<any>) => {
+const inject = (identifier?: interfaces.ServiceIdentifier<unknown>) => {
     return function(target: Vue, prop: string) {
         // Try use reflection to generate the identifier if not provided explicitly.
         if (!identifier && typeof(Reflect) !== 'undefined' && typeof(Reflect.getMetadata) === 'function')
-            identifier = Reflect.getMetadata('design:type', target, prop);
+            identifier = Reflect.getMetadata('design:type', target, prop) as interfaces.ServiceIdentifier<unknown>;
 
         if (!identifier)
             throw new TypeError('Unable to resolve identifier; please specify it explicitly');
 
         const injectable = (typeof(target) === 'function' ? target : target.constructor) as InjectableVueClass;
-        const $inject = injectable.$inject || (injectable.$inject = {});
+        const options = injectable.options || (injectable.options = {});
+        const dependencies = options.dependencies || (options.dependencies = {});
 
-        $inject[prop] = identifier;
+        dependencies[prop] = identifier;
     }
 };
 
+function getContainer(component: Vue): interfaces.Container | undefined {
+    if (component.$options.container)
+        return component.$options.container;
+    if (component.$parent)
+        return getContainer(component.$parent);
```

Comment:
I don't know how deep Vue's component trees get but could we trivially do this with a `while` loop instead of recursion?

```js
const current = component;
let container: interfaces.Container | undefined;

while (current != null && container == null) {
    container = current.$options.container;
    current = current.$parent;
}

return container;
```

Reply:
yep good call, with the attaching the found one to itself it should limit the search depth of even deep component trees, unless you have smart leaves deep down under a lot of dumb presentation and layout components